crates/wasmparser/src/validator/core.rs

```diff
@@ -493,56 +496,77 @@ pub(crate) struct Module {
 }
 
 impl Module {
-    pub fn add_type(
+    pub fn add_types(
         &mut self,
-        ty: SubType,
+        types_added: Vec<SubType>,
```

Comment:
I think this can be a slice, rather than a `Vec` which forces heap allocations on callers, particularly callers that are just passing in a single type.

```suggestion
        types_added: &[SubType],
```

And then single-type callers can use https://doc.rust-lang.org/nightly/std/slice/fn.from_ref.html

Reply:
Done.